src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1206,12 +1211,12 @@ CompactionAggregate.UnifiedAggregate constructAggregate(Controller controller, l
             {
```

Comment:
The `maxSSTablesToCompact` calculation above is bad. Because a bucket may expand horizontally (due to overlap inclusion), the size of individual sstables (i.e. `level.avg`) is not the right divisor here.

I don't know how to sensibly improve it, though. Hopefully STAR-1872/CNDB-11016 can provide a different solution to this problem.